server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -66,6 +66,7 @@
  * Note: This class is performance sensitive, so we pay extra attention on the data structure usage and we avoid streams and iterators
  * when possible in favor of the classic for-i loops.
  */
+@SuppressWarnings("ForLoopReplaceableByForEach")
```

Comment:
Isn't this just a warning in your IDE? I assumed we only suppress warnings in the code that actually block the compiler (like `this-escape` etc). In your IDE, you can just disable inspection yourself if it bothers you. I also don't see any other usages of this warning suppression - not that that on its own is a reason not to add it, but still.

Reply:
FWIW, what @nielsbauman says was my assumption as well.

Reply:
That is true, this is not absolutely necessary addition. 

I have been going back and forth for this, but in the end I thought there is value here for the following two reasons (considering a lot of us use the IDE):
- It reduces the warning fatigue of the dev that uses the IDE.
- It makes it a more conscious action to go and change a for loop to foreach, instead of just typing a shortcut.

I wasn't aware that we were actively against suppressing such warnings. If we are I am happy to revert.